### ISSUE TITLE
Ensure xlwings communicates with a specific workbook

### ISSUE THREAD

**Issue by Matt-Greer** [xlwings/xlwings#2243]
I apologize for the length of this.  I appreciate any and all advice you can give me.  Thank you!

#### OS
Windows 10

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
Excel 365 (Version 2302)
Python v3.9.13
xlwings v0.27.11 (I am "forced" to use this version)

I'll put the questions first as the background probably is necessary, but I'll leave it at the end anyway.

#### Questions
**tl;dr** I want to be able to work in another Excel spreadsheet without conflicting with an existing spreadsheet that's exchanging data with python.

1. What is the proper method to associate an excel workbook to the running python program?  I tried to understand how to use the context manager, but it's not really feasible I don't think.  I'm using a python server (it's just a command window that runs on my laptop) that "listens" for messages sent from a subroutine in Excel.  I didn't write that client/server code nor can I edit it.  So I think I have to stick to working with a global variable reference to the workbook.
2. Likewise, I tried to understand `xw.apps.keys` but I just couldn't figure out how to associate a key with a workbook name, and if I have the workbook name I just may as well use that I suppose. 

#### Background
Up until recently, the python program I wrote would simply refer to the active workbook and specific named ranges within that workbook. (I fully acknowledge that I may be doing many things wrong as I am a beginner.)  So, for example:

```python
STUDIES_TBL       = "tblStudies"            # Excel table name
STUDIES_TBL_ALL   = "tblStudies[[#All]]"    # Excel full table reference including headers
STUDIES_TBL_START = "SensitivityStudy"      # Start of the table, first column, header row

wb = xw.books.active

# Get a dataframe from an Excel table
df: pd.DataFrame = xw.Range(STUDIES_TBL_ALL).options(pd.DataFrame, index=0, header=True).value

# Get a list of values not in a table, but the list has a header
src_list: list[str] = xw.Range(SOURCE).offset(1,0).expand('down').value

# Get a value from a named range
firstHeader = xw.Range(STUDIES_TBL_START).value
```
I started researching on how to have python/xlwings "know" the exact workbook with which it should be exchanging data and I've begun making changes.  What I've come to understand that I need to do is assign the variable `wb` as follows.  (I know I can use the actual name of the workbook, which I do in my program.)

```py
wb: xw.Book = xw.books[0]

# And then to refer to a named range:
src_list: list[str] = wb.names[SOURCE).refers_to_range.offset(1,0).expand('down').value

# To get a dataframe from a table
table: xw.main.Table = wb.sheets[ws].tables[STUDIES_TBL]
df: pd.DataFrame = pd.DataFrame(table.range.value)
 
```
Thank you again for your time and consideration,

-matt

**Comment by fzumstein** [comment]
Yes, `xw.Range` uses the active sheet of the active book of the active Excel instance (app). I've tried to get rid of all examples that use this syntax in the docs though.

Your second code snippet looks good? What's the issue with it? You should probably use `xw.Book("your-name.xlsx")` instead of `xw.books[...]` but that's about it (your current version looks only in the active instance of Excel).

If you had control over the backend, you'd use `xw.Book.caller()` instead though.

**Comment by Matt-Greer** [comment]
As I respond, I cannot thank you enough for your reply and review of what I wrote.  I crave feedback from experts such as yourself and wouldn't have gotten as far as I have without it.  So again, thank you!

Before the history behind the post... I am unfamiliar with `xw.Book.caller()` so I'll research how to use that later today.

The workbook I'm developing has a specific string in the name (but the name is not just that string) so I'm using that string in the script to determine which workbook, if there's more than one open, to "link" to python:
```py
num_books: int = xw.books.count

if xw.books.count == 1:
    wb: str = xw.books[0]
elif xw.books.count > 1:
    wb = None
    for i in range(num_books):
        book_name = xw.books[i].name
        if 'MYSPECIFICSTRING' in book_name:
            wb = xw.books[i]
            break
else:
    print("There's no open workbook... can I get an F in the chat?")
```

So what happened was my original message was much longer and between starting the message and then making sure I didn't ask you questions I couldn't answer myself I figured out how to get a dataframe from an Excel table without using `xw.Range...`.  That was the main original reason I was writing;  while implementing the changes to using `refers_to_range` I got stuck on a part where I was pulling in an Excel structured table into a dataframe.  I couldn't get the syntax correct.

Originally I was using this to get a dataframe from a structured table:
```py
df = xw.Range(myTable + "[[#All]]").options(pd.DataFrame, index=0, header=True).value
```

I kept searching the documentation for references to 'listobject' but couldn't find anything that helped me get the syntax correct.  But eventually I did figure out what you saw in my original post.

If you don't mind, may I leave this thread open in case I have questions about using `xw.Book.caller()`?  If you wish to close it I understand.

Thanks again,

-Matt
